pandas/src/inference.pyx

```diff
@@ -569,18 +569,17 @@ def maybe_convert_numeric(object[:] values, set na_values,
             seen_complex = True
         else:
             try:
-                status = floatify(val, &fval)
+                status = floatify(val, &fval, &maybe_int)
                 floats[i] = fval
                 if not seen_float:
-                    if '.' in val or fval == INF or fval == NEGINF:
-                        seen_float = True
-                    elif 'inf' in val:  # special case to handle +/-inf
-                        seen_float = True
-                    elif fval < fINT64_MAX and fval > fINT64_MIN:
-                        try:
-                            ints[i] = int(val)
-                        except ValueError:
-                            ints[i] = <int64_t> fval
+                    if maybe_int:
+                        if fval < fINT64_MAX and fval > fINT64_MIN:
+                            try:
+                                ints[i] = int(val)
+                            except ValueError:
```

Comment:
It produces a string in that case:
``` .python
In [4]: pd.read_csv(StringIO(str(np.iinfo(np.int64).max + 1)), header=None, engine='python').values
Out[4]: array([['9223372036854775808']], dtype=object)

In [5]: pd.read_csv(StringIO(str(np.iinfo(np.int64).min - 1)), header=None, engine='python').values
Out[5]: array([['-9223372036854775809']], dtype=object)
```

Do you want it to infer the dtype ``uint64``? The C parser doesn't do that either currently.

I don't see how that line ``ints[i] = int(val)`` could ever throw an exception, though. Maybe there was some case that's now been eliminated?